core/modules/catalog/store/product/actions.ts

```diff
@@ -664,54 +646,27 @@ const actions: ActionTree<ProductState, RootState> = {
+        context.commit(types.PRODUCT_SET_GALLERY, attributeImages(context.state.current))
       } else {
         let productGallery = uniqBy(configurableChildrenImages(product).concat(getMediaGallery(product)), 'src').filter(f => { return f.src && f.src !== config.images.productPlaceholder })
-        context.commit(types.CATALOG_UPD_GALLERY, productGallery)
+        context.commit(types.PRODUCT_SET_GALLERY, productGallery)
       }
     } else {
       let productGallery = uniqBy(configurableChildrenImages(product).concat(getMediaGallery(product)), 'src').filter(f => { return f.src && f.src !== config.images.productPlaceholder })
-      context.commit(types.CATALOG_UPD_GALLERY, productGallery)
+      context.commit(types.PRODUCT_SET_GALLERY, productGallery)
     }
   },
 
   /**
    * Load the product data - async version for asyncData()
    */
-  fetchAsync (context, { parentSku, childSku = null, route = null }) {
-    if (context.state.productLoadStart && (Date.now() - context.state.productLoadStart) < PRODUCT_REENTER_TIMEOUT) {
-      Logger.log('Product is being fetched ...', 'product')()
-    } else {
-      context.state.productLoadPromise = new Promise((resolve, reject) => {
-        context.state.productLoadStart = Date.now()
-        Logger.info('Fetching product data asynchronously', 'product', {parentSku, childSku})()
-        EventBus.$emit('product-before-load', { store: rootStore, route: route })
-        context.dispatch('reset').then(() => {
-          context.dispatch('fetch', { parentSku: parentSku, childSku: childSku }).then((subpromises) => {
-            Promise.all(subpromises).then(subresults => {
-              EventBus.$emitFilter('product-after-load', { store: rootStore, route: route }).then((results) => {
-                context.state.productLoadStart = null
-                return resolve()
-              }).catch((err) => {
-                context.state.productLoadStart = null
-                Logger.error(err, 'product')()
-                return resolve()
-              })
-            }).catch(errs => {
-              context.state.productLoadStart = null
-              reject(errs)
-            })
-          }).catch(err => {
-            context.state.productLoadStart = null
-            reject(err)
-          }).catch(err => {
-            context.state.productLoadStart = null
-            reject(err)
-          })
-        })
-      })
-    }
-    return context.state.productLoadPromise
+  async fetchAsync (context, { parentSku, childSku = null, route = null }) {
+    Logger.info('Fetching product data asynchronously', 'product', {parentSku, childSku})()
+    EventBus.$emit('product-before-load', { store: rootStore, route: route })
+    await context.dispatch('reset')
+    const product = await context.dispatch('fetch', { parentSku: parentSku, childSku: childSku })
+    await EventBus.$emitFilter('product-after-load', { store: rootStore, route: route })
+    return product
   }
 }
 
```

Comment:
do we still need `fetchAsync` if it got so much simpler? I think it's pretty much misleading with `fetch` and should be reduced to single method

Reply:
good point, will change that!